internal/fnruntime/container.go

```diff
@@ -197,6 +197,7 @@ func (f *ContainerFn) prepareImage() error {
 	// This can help to ensure we have the latest release for "moving tags" like
 	// v1 and v1.2. The performance cost is very minimal, since `docker pull`
 	// checks the SHA first and only pull the missing docker layer(s).
+	f.Image = AddDefaultImagePathPrefix(f.Image)
```

Comment:
We probably want to resolve this at the higher layer. Look where we create `containerRunner` and `functionRunner` in `runner.go` because there might be other pieces where full name is probably desired ?

Reply:
Moved it up.

Reply:
I looked at the code. There are two places where the changes will ensure that they are propagated in all places:

- for `fn eval`, we should resolve this in the command runner itself like we are doing in the `cmdfndoc.go`
- for `fn render`, doing it in `fnChain` function which will ensure it's propagated in all places. Another option is doing it in `pkg.Pipeline()` function.

Calling it individually where we make calls to docker or displayResults is not safe IMO.